### PR TITLE
Fix build after moving classes to the other module in upstream

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### What does this PR do?
Fix build after moving classes to the `infrastrucure-kubernetes` maven module in Eclipse Che.
https://github.com/eclipse/che/pull/17457

### CODE REVIEW

**Comment by ibuziuk** [comment]
@vparfonov lets's merge after green PR check

**Comment by vparfonov** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

File: plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/Fabric8MultiTenantModule.java [hunk 1]
```diff
@@ -29,7 +29,6 @@
   protected void configure() {
     LOGGER.info("Configuring {}", this.getClass().getName());
     bind(OpenShiftClientFactory.class).to(Fabric8OpenShiftClientFactory.class);
-    bind(KubernetesClientFactory.class).to(Fabric8OpenShiftClientFactory.class);
```
Comment (ibuziuk) [inline plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/Fabric8MultiTenantModule.java:32]: I believe we can not remove this binding

Comment (vparfonov) [reply plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/Fabric8MultiTenantModule.java:32]: Added here https://github.com/eclipse/che/pull/17457/files#diff-e1932b1dcac1cd050ac5e68ca54d482fR118

Comment (ibuziuk) [reply plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/Fabric8MultiTenantModule.java:32]: yeah, but the problem is that we need to use Fabric8OpenShiftClientFactory for Hosted Che - https://github.com/redhat-developer/rh-che/blob/master/plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/Fabric8OpenShiftClientFactory.java#L38

**Comment by vparfonov** [comment]
[test]

**Pull request merged**